File: CodeEval/medium/046primenumbers/solution.js
#!/usr/bin/env node

function write(s) {
    process.stdout.write(s);
}

function isprime(n) {
    if (n % 2 === 0) return false;
    var sqt = parseInt(Math.sqrt(n));
    for (var i = 3; i <= sqt; i += 2) {
        if (n % i === 0) return false;
    }
    return true;
}

function primes(n) {
    if (n > 2) write("2");
    for (var i = 3; i < n; i += 2) {
        if (isprime(i)) write("," + i.toString());
    }
    write("\n");
}

require("fs").readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
    line = line.trim();
    if (line !== "") {
        primes(parseInt(line));
    }
});

File: CodeEval/medium/046primenumbers/solution.rb
#!/usr/bin/env ruby

def isprime(n)
  return false if n % 2 == 0
  (3..Math.sqrt(n).to_i).step(2) { |i| return false if n % i == 0 }
  return true
end

def primes(n)
  (3...n).step(2) { |i| yield i if isprime(i) }
end

File.open(ARGV[0]).each_line do |line|
  n = line.to_i
  if n > 2
    print 2
    primes(n) { |i| print ",", i }
  end
  puts
end
